packages/shared/src/hooks/useCreateSquadModal.ts

```diff
@@ -24,26 +27,12 @@ export const useCreateSquadModal = ({
 }: UseCreateSquadModalProps): UseCreateSquadModal => {
   const router = useRouter();
   const { openModal } = useLazyModal();
-  const previousRef = useRef(null);
   const [hasTriedOnboarding, setHasTriedOnboarding, isLoaded] =
     usePersistentContext<boolean>(SQUAD_ONBOARDING, hasSquads);
 
-  const openNewSquadModal = () =>
-    openModal({
-      type: LazyModal.NewSquad,
-      props: {
-        onPreviousState: () => previousRef.current(),
-      },
-    });
-  const openSquadBetaModal = () =>
-    openModal({
-      type: LazyModal.NewSquad,
-      props: {
-        shouldShowIntro: true,
-        onPreviousState: () => previousRef.current(),
-      },
-    });
-  previousRef.current = openSquadBetaModal;
+  const openNewSquadModal = (props: ModalProps = {}) =>
+    openModal({ type: LazyModal.NewSquad, props });
+  const openSquadBetaModal = () => openNewSquadModal({ shouldShowIntro: true });
```

Comment:
Just wait with merging please, I revamped this for analytics 😅

Reply:
Oh, sure thing. No worries.

Reply:
@sshanzel I included these changes in my tracking event :)